llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -10272,6 +10288,9 @@ void llvm::findValuesAffectedByCondition(
                                                            m_Value()))) {
       // Handle patterns that computeKnownFPClass() support.
       AddAffected(A);
+    } else if (!IsAssume && match(V, m_CombineOr(m_Trunc(m_Value(X)),
```

Comment:
Is `!IsAssume` used to avoid issues with ephemeral values? If so, please add some comments here.

Reply:
partly, for the not https://github.com/llvm/llvm-project/pull/126423#discussion_r1948229739 it is the avoid issues with ephemeral values. 
but also to avoid to add the same value again as truncated value of V is already added for assume above as the AddAffected Looks through trunc.
will add comment